### ISSUE TITLE
When playing a track from a Jingle mode playlist the previous songs metadata is shown

### ISSUE THREAD

**Issue by nunotfc** [AzuraCast/AzuraCast#4774]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

5500549 (2021-11-09 14:42) • Docker • PHP 8.0

### What happened?

even with the option checked in playlists, the metadata of the anouncement, gingles are being sent.

**Comment by nunotfc** [comment]
update: sometimes the gingle mode work, but:
but when gingle is playing the metadata is the song before that the ended for exammple:
i have song01, song02, gingle
song01 plays ok, song 02 plays ok. when play gingle, the metadata goes back to song01 until the gingle ends.

**Comment by Vaalyn** [comment]
I haven't yet seen it showing the metadata of a Jingle when having the `Hide Metadata from Listeners ("Jingle Mode")` option active but I can see it briefly showing the metadata of the previous song while the Jingle is running.

I think this is most likely related to the changes to the `NowPlayingApiGenerator` for the new prefetching of songs.
Will need to investigate the exact reason of it.

Thank you for the report.

**Comment by nunotfc** [comment]
hello, what about this bug? any news?

**Comment by SlvrEagle23** [comment]
@nunotfc Re-reading through this issue and I'm struggling to clarify what the actual bug being reported is. Are you reporting that it's a bug that when Jingle playlists are played, the previous song's metadata is played instead, or that this _isn't_ happening?

Basically, the whole point of "Jingle Mode" playlists is that they hide their metadata from the end user. In our internal implementation of that, we just repeat the metadata of the previous track to accomplish that.

If that's not working as we designed it, we can look into it, but otherwise that is how it is intended to work.

**Comment by nunotfc** [comment]
yes lets take a senario.
i have rihanna humbrela, metalika nothing else mathers and a gingle.
rihanna is played, after metalika is played. when a jplaylist that have jingle mode enabled plays, the metadata change 2 3 times.
when jingle is played i see the metadata of rihanna humbrela, when jingle ends i see metalika for 11 2 seconds and finally i see the next correct song.

**Comment by SlvrEagle23** [comment]
@nunotfc Thank you for the detailed response. I looked into it and I was able to confirm that jingle mode playlists are incorrectly trying to mimic the last song that was _played_, not the song that was ahead of them in the queue. This was fine in our previous way of working, but now that Liquidsoap maintains its own small internal queue, it causes trouble. I've updated how we handle Jingle Mode playlists, and that version is currently deploying (see the commit above). Once that's deployed, let me know if updating to the newest version helps resolve things on your end.
